hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmMetadataManagerImpl.java

```diff
@@ -1571,11 +1560,33 @@ public List<BlockGroup> getPendingDeletionKeys(final int keyCount,
             }
             currentCount++;
           }
-          keyBlocksList.addAll(blockGroupList);
+
+          List<OmKeyInfo> notReclaimableKeyInfoList =
+              notReclaimableKeyInfo.getOmKeyInfoList();
+
+          // If all the versions are not reclaimable, then do nothing.
+          if (notReclaimableKeyInfoList.size() > 0 &&
```

Comment:
3 case, not getting if all scenario handled:
1. all versions are not reclaimable - handled
2. all versions are reclaimable - not getting how handled
3. blocks are added if few blocks are reclaimable

Reply:
For case 2, if all versions are reclaimable then notReclaimableKeyInfoList will be 0. So it will be added to `keyBlocksList` and it will be deleted by the PurgeRequest 
```
 if (notReclaimableKeyInfoList.size() !=
      infoList.getOmKeyInfoList().size()) {
    keyBlocksList.addAll(blockGroupList);
  }
```
3) I don't see a case where we only delete a few blocks in `OMKeyInfo`. Let me know if there is some case.